string uri
int32 encoder_position
---
string uri
int32 encoder_position